### DISCUSSION TITLE
Trying to make theme change dynamically

### DISCUSSION THREAD

**Discussion by LamprosPitsillos** [qtile/qtile#3950] [Q&A]
I have set up this:
```
├──  config.py
├──  current_theme.py
├──  current_wallpaper.py
└──  themes
   ├──  One-Dark.py
   └──  One-Light-custom.py
```
```python
from current_theme import THEME
from current_wallpaper import WALLPAPER
```
And i change the current_*.py using shell scripts and also sending USR1 to update the config.

**Is there a better way?**

**The only problem i have is that the `nm-applet` doesn't change the icons from light to dark because i don't know how to do this dynamically... Is there a way to do it?**

PS' the script if anyone wants something similar
```shell
#!/bin/sh

case "$1" in
    LIGHT)
        cat ~/.config/kitty/One-light.conf > ~/.config/kitty/current-theme.conf
        cat ~/.config/qtile/themes/One-Light-custom.py> ~/.config/qtile/current_theme.py
        ;;
    DARK)
        cat ~/.config/kitty/One-dark-custom.conf > ~/.config/kitty/current-theme.conf
        cat ~/.config/qtile/themes/One-Dark.py > ~/.config/qtile/current_theme.py
        ;;
    *) echo default
        ;;
esac
# Send signal to reload configs
pkill -USR1 kitty
pkill -USR1 qtile

```

**Comment by m-col** [comment]
nm-applet is a GTK app, and I found this discussion suggesting you can set GTK dark/light theme on a per-app basis with some environmental variables: https://unix.stackexchange.com/questions/14129/gtk-enable-set-dark-theme-on-a-per-application-basis. From what I can tell, this might do it (in your shell script):
```sh
pgrep nm-applet && pkill nm-applet
GTK_THEME=:dark nm-applet &
```
